dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategorySQL.java

```diff
@@ -40,5 +40,9 @@ public String getSortedChildren() {
 				"category.inode = tree.child and tree.parent = ? and category_1_.inode = category.inode " +
 				" and category_1_.type = 'category'";
 	}
+	
+	public String getVelocityVarNameCount() {
+	    return "select count(*) as test from category where category_velocity_var_name like ?";
+	}
```

Comment:
What @jdotcms said. There's a chance that the whole CategoryFactory implementation can see a refactoring in the same way we did for content types (mostly for SQL operations), and current logic here (which it's the same for fields and content types) was missing for categories, which also require a VelVarName once they're created